htdocs/inc.processCheckCardEditRegister.php

```diff
@@ -171,6 +171,21 @@ function fillRfidArrAvailWithUsed($rfidAvailArr, $rfidUsedArr=array()) {
         $post['audiofolderNew'] = $link = str_replace(array('http://','https://','/','=','-','.', 'www','?','&'), '', $post['streamURL']);
     }
 
+    //wrong spotify url, convert to mopidy format
+    if((isset($post['streamURL']) && $post['streamType'] == "spotify") && (strpos($post['streamURL'], "https://open.spotify.com/") !== false)){
+        $messageError .= "Wrong spotify url, converted to the correct format";
+        $patterns = array();
+        $patterns[0] = '/https\:\/\/open.spotify.com/';
+        $patterns[1] = '/\/(playlist|album|track|artist)\//';
+        $patterns[2] = '/(\w+)\?(.*)/';
+        $replacements = array();
+        $replacements[0] = 'spotify:';
+        $replacements[1] = '$1:';
+        $replacements[2] = '$1';
+        $post['streamURL'] = preg_replace($patterns, $replacements, $post['streamURL']);
+    }
```

Comment:
Maybe some kind of error checking and a log error would be helpful, if Spotify changes the URL format? So we would have at least an error message that the conversion doesn't work anymore?

Reply:
Hmmm, the replacement itself is a error handling already. The ideal way would be a check on the correct mopidy format and preventing inserting wrong urls into the "database"/filesystem.

Reply:
For the moment this would be ok for me. Maybe you could add a log entry with the content of the new URI, so in the future we would have a log entry, if the conversion had an error.

Reply:
Sure, I can do that. You mean a debug.log entry with the setting DEBUG_WebApp = on?

Reply:
Yes. Thanks

Reply:
I had another idea and improved and translated the shown error message. Would this be ok?
![image](https://user-images.githubusercontent.com/75278644/103698728-10cb7e80-4fa2-11eb-9e1d-0e15eb4bf6e2.png)